ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -153,6 +165,13 @@ a database ( directory )
           partitionsPostProcessing(iterator, scope, loadTaskTracker, tableTracker,
               partitionsTracker);
           partitionsTracker.debugLog("partitions");
+
+          if (!loadTaskTracker.hasReplicationState()) {
+            // Add ReplStateLogTask only if no pending table load tasks left for next cycle
+            ImportTableDesc tableDesc = loadPartitions.tableDesc();
+            loadTaskTracker.updateTaskCount(createTableReplLogTask(partitionsTracker.tasks(),
```

Comment:
can we use addTask method rather ? and not have it exposed too, also these tasks are to be tracked, also where is the task returned being added to the parent graph ? these should be